incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -67,3 +76,26 @@ func getSelectorFromString(str string) (labels.Selector, error) {
 	}
 	return selector, nil
 }
+
+// GetNoneSelector returns true indicates that user do not want this object to be propagated
+func GetNoneSelector(inst *unstructured.Unstructured) (bool, error) {
+	noneSelectorStr := GetNoneSelectorAnnotation(inst)
+	// Empty string is treated as 'false'. In other selector cases, the empty string is auto converted to
+	// a selector that matches everything.
+	if noneSelectorStr == "" {
+		return false, nil
+	}
+	noneSelector, err := strconv.ParseBool(noneSelectorStr)
+	if err != nil {
+		// When the user put an invalid noneSelector, we choose not to propagate this object to any child
+		// namespace to protect any object in the child namespaces to be overwritten
+		return true,
+			fmt.Errorf("invalid %q value: %q, %w",
```

Comment:
Are you sure that `ParseBool` doesn't include the bad string? If you are, then I'd suggest: "invalid %s value %q: %w"

What you have now will print something like:

`invalid "propagate.hnc.x-k8s.io/none" value: "ture", not a boolean string`

The initial double quotes aren't necessary because the annotation string won't be mistaken for a regular English word, and having a comma between the bad value and the error message isn't a strong separation between the different parts of the error. Instead, I'm suggesting: 

`invalid propagate.hnc.x-k8s.io/none value "ture": not a boolean string`

Now we're reserving the double quotes for the only part that could potentially be mistaken as an English word, and there's a strong separation between the error portions. Make sense?

Reply:
Yeah I see, updated the comments. `ParseBool` does include the bad string so I removed it from our log